dspace-api/src/test/java/org/dspace/content/logic/LogicalFilterTest.java

```diff
@@ -0,0 +1,358 @@
+     */
+    @Before
+    @Override
+    public void init() {
+        super.init();
+        try {
+            context.turnOffAuthorisationSystem();
+            // Set up logical statement lists for operator testing
+            setUpStatements();
+            // Set up DSpace resources for condition and filter testing
+            this.owningCommunity = communityService.create(null, context);
+            this.collection = collectionService.create(context, owningCommunity);
+            WorkspaceItem workspaceItem = workspaceItemService.create(context, collection, false);
+            this.itemOne = installItemService.installItem(context, workspaceItem);
+            workspaceItem = workspaceItemService.create(context, collection, false);
+            this.itemTwo = installItemService.installItem(context, workspaceItem);
+            // Initialise metadata field for later testing with both items
+            this.metadataField = metadataFieldService.findByElement(context,
+                MetadataSchemaEnum.DC.getName(), element, qualifier);
+            context.restoreAuthSystemState();
+        } catch (AuthorizeException ex) {
+            log.error("Authorize Error in init", ex);
+            fail("Authorize Error in init: " + ex.getMessage());
+        } catch (SQLException ex) {
+            log.error("SQL Error in init", ex);
+            fail("SQL Error in init: " + ex.getMessage());
+        }
+    }
+
+    /**
+     * This method will be run after every test as per @After. It will
+     * clean resources initialized by the @Before methods.
+     *
+     * Other methods can be annotated with @After here or in subclasses
+     * but no execution order is guaranteed
+     */
+    @After
+    @Override
+    public void destroy() {
+        context.turnOffAuthorisationSystem();
+        // Delete resources
+        try {
+            itemService.delete(context, itemOne);
+            itemService.delete(context, itemTwo);
+            itemService.delete(context, itemOne);
+            collectionService.delete(context, collection);
+            communityService.delete(context, owningCommunity);
+        } catch (Exception e) {
+            // ignore
+        }
+        context.restoreAuthSystemState();
+
+        // Set all class members to null
+        owningCommunity = null;
+        collection = null;
+        itemOne = null;
+        itemTwo = null;
+        trueStatements = null;
+        trueFalseStatements = null;
+        falseStatements = null;
+        trueStatementOne = null;
+        falseStatementOne = null;
+        element = null;
+        qualifier = null;
+        metadataField = null;
+
+        super.destroy();
+    }
+
+    /**
+     * Test the AND operator with simple lists of logical statements
+     */
+    @Test
+    public void testAndOperator() {
+        // Blank operator
+        And and = new And();
+        // Try tests - the item can be null, as the statements are simply returning booleans themselves
```

Comment:
I disagree here as null is a very special value and we are hardcoding in the test the logic of our current implementation. Any statement about a null item could/should return false.
Can you switch the test to use a real item?

Reply:
the comment here and in the other test about the fact that the item can be null should be removed

Reply:
Understood - I fixed the comment and also used itemOne as a test case for all operator tests just to make it clearer that there is an item passed, and avoid any random null handling as you requested